sygnal/webpushpushkin.py

```diff
@@ -336,25 +348,15 @@ def post(self, endpoint, data, headers, timeout):
             timeout (int)
                 Ignored for now
         """
-        body_producer = FileBodyProducer(BytesIO(data))
-        # Convert the headers to the camelcase version.
-        headers = {
-            b"User-Agent": ["sygnal"],
-            b"Content-Encoding": [headers["content-encoding"]],
-            b"Authorization": [headers["authorization"]],
-            b"TTL": [headers["ttl"]],
-        }
-        deferred = self.http_agent.request(
-            b"POST",
-            endpoint.encode(),
-            headers=Headers(headers),
-            bodyProducer=body_producer,
-        )
-        return HttpResponseWrapper(deferred)
+        return HttpDelayedRequest(endpoint, data, headers)
 
 
-class HttpResponseWrapper:
+class HttpDelayedRequest:
     """
+    Captures the values received from pywebpush for the endpoint request.
+    The request isn't immediately executed, to allow adding headers
+    not supported by pywebpush, like Topic and Urgency.
+
     Provide a response object that matches the API expected from pywebpush.
     pywebpush expects a synchronous API, while we use an asynchronous API.
```

Comment:
this looks outdated?

Reply:
This still serves as something that looks like a response to pywebpush, while actually containing the information to perform the request ourselves once pywebpush finishes. I agree the naming is not ideal, but couldn't come up with something better. Ideas?

Reply:
right, I see.